tests/manage/monitoring/conftest.py

```diff
@@ -0,0 +1,142 @@
+        """
+        Log all alerts from Prometheus API every 10 seconds.
+
+        Args:
+            run (bool): When this var turns into False the thread stops.
+            alert_list (list): List to be populated with alerts
+        """
+        prometheus = PrometheusAPI()
+        while info.get('run'):
+            alerts_response = prometheus.get(
+                'alerts',
+                payload={
+                    'silenced': False,
+                    'inhibited': False
+                }
+            )
+            assert alerts_response.ok, 'Prometheus API request failed'
+            for alert in alerts_response.json().get('data').get('alerts'):
+                if alert not in alert_list:
+                    logger.info(f"Adding {alert} to alert list")
+                    alert_list.append(alert)
+            time.sleep(10)
+
+    if not measure_after:
+        start_time = time.time()
+
+    # init logging thread that checks for Prometheus alerts
+    # while workload is running
+    # based on https://docs.python.org/3/howto/logging-cookbook.html#logging-from-multiple-threads
+    info = {'run': True}
+    alert_list = []
+
+    logging_thread = threading.Thread(
+        target=prometheus_log,
+        args=(info, alert_list)
+    )
+    logging_thread.start()
+
+    try:
+        result = operation()
+        if measure_after:
+            start_time = time.time()
+        passed_time = time.time() - start_time
+        if minimal_time:
+            additional_time = minimal_time - passed_time
+            if additional_time > 0:
+                time.sleep(additional_time)
+        stop_time = time.time()
+    except KeyboardInterrupt:
+        # Thread should be correctly terminated on next few lines
+        # this is done in case of user interuption to make sure that thread
+        # is terminated correctly
+        pass
+    info['run'] = False
+    logging_thread.join()
+    logger.info(f"Alerts found during measurement: {alert_list}")
+    return {
+        'start': start_time,
+        'stop': stop_time,
+        'result': result,
+        'metadata': metadata,
+        'prometheus_alerts': alert_list
+    }
+
+
+@pytest.fixture(scope="session")
+def workload_stop_ceph_mgr():
+    """
+    Downscales Ceph Manager deployment, measures the time when it was
+    downscaled and monitors alerts that were triggered during this event.
+
+    Returns:
+        dict: Contains information about `start` and `stop` time for stopping
+            Ceph Manager pod.
+    """
+    oc = ocp.OCP(
+        kind=constants.DEPLOYMENT,
+        namespace=config.ENV_DATA['cluster_namespace']
+    )
+    mgr_deployments = oc.get(selector=constants.MGR_APP_LABEL)['items']
+    mgr = mgr_deployments[0]['metadata']['name']
+
+    def stop_mgr():
+        """
+        Downscale Ceph Manager deployment for 6 minutes. First 5 minutes
+        the alert should be in 'Pending'.
+        After 5 minutes it should be 'Firing'.
```

Comment:
If you have a reference (documentation, requirement or code) for the 5 minute interval, it would be nice to have it added there.

Reply:
done